### ISSUE TITLE
[ALM Accelerator - QUESTION] Pipeline Variables & Templates

### ISSUE THREAD

**Issue by ChristopheHare** [microsoft/coe-starter-kit#7202]
### What is your question?

Hi team,

I have a Question regarding the Pipeline Variables & their correlation to the $Variables within the templates.

I may have misunderstood from documentation, but i am currently of the thought that when adding the variable at to the pipeline, it should be picked up by the templates, or do i need to "hardcode" the templates for my desired outcomes?

The below image shows 3 Variables,
The Service Connection variable is the only one being picked up by the pipeline.
![image](https://github.com/microsoft/coe-starter-kit/assets/118164862/f3338f38-3fe8-4496-bc93-db6c8b08c488)

environmentName - to my knowledge should be the DevOps environment? Or am i wrong? as shown below none of the values i have placed within the pipeline variable have worked.
![image](https://github.com/microsoft/coe-starter-kit/assets/118164862/1d5a3dbe-4b3d-4bd0-b144-3c65dd66b4e2)

owner.OwnerEmail - attempting to set ownership either to myself or Service Account, but still the Service Principle (App Reg/User) is granted ownership despite the value passed in the variable.

Appreciate, this is most likely user error. But would very much appreciate the help, i've tried to search through old issues and googling what i could be doing wrong, but not been able to find anything. 

Thanks in advance, Much appreciated.

### What component are you experiencing the issue with?

ALM Accelerator Pipelines

### What solution version are you using?

Sept 23

AB#1616

**Comment by RajeevPentyala** [comment]
@ChristopheHare To start with basics, 

- When you commit a solution for the first time, it will create a certain number of deployment pipelines, depending on the number of 'Deployment Environments' you have configured in your 'Solution Profile'.

![image](https://github.com/microsoft/coe-starter-kit/assets/104883923/affe68e1-fb58-4d46-b202-c98d99ec1b48)

- Every deployment pipeline created during the solution commit, by default, receives a variable named 'EnvironmentName,' with the value set to the name of the respective 'Deployment Environment.'

- Additionally, the solution commit creates a folder 'config' along with subfolders with environment names in the Azure DevOps repository.
![image](https://github.com/microsoft/coe-starter-kit/assets/104883923/58caa9ce-7dc4-41ef-96f8-8e6f513fad26)

- When deploying the solution, the deployment pipeline reads the value passed in the 'EnvironmentName' variable and then retrieves the content of 'customDeploymentSettings.json' under 'config/{environmentname}/customDeploymentSettings.json.'
![image](https://github.com/microsoft/coe-starter-kit/assets/104883923/521746be-8e5b-41c3-b50d-a3a087581447)

In summary, it's essential to note that you cannot pass any arbitrary value in the 'EnvironmentName'; it must correspond to a valid 'Deployment Environment' configured in your 'Solution Profile.' Similarly, the value for 'owner.OwnerEmail' depends on the specified action, such as setting the owner of Canvas Apps, Flows, etc.

I hope this information is clear. Feel free to reach out if you have any further queries.

**Comment by ChristopheHare** [comment]
Thanks for the support and prompt response Rajeev,

Hope you are well,

I will conduct some further tests on this, but after my test Friday, no deployment pipelines were created upon the first commit. subsequently, when i tried a deployment from ALMA a PR was created, but no build was queued.
previously i had copied the .yaml and created the _build-deploy-_SolutionNameHere_.yml files within the solution folder in the solution repository and referenced those within the pipeline, along with the alm-accelerator-templates repository, in line with the feedback from Mike on our previous call. Is this something i still need to do? If your saying it should be taken care of by the first commit, what is it i have missed? 
This has lead to successful deployments, but the App User (AAD App Reg within the Application User in environment) is listed as owner.

However; The details entered within Configure deployment settings have come through as expected. :) in both the custom and deploymentsettings.json files respectively.

The advice on learn.microsoft.com states 

"Every deployment pipeline requires the environment variable EnvironmentName and the service connection variable ServiceConnection.

EnvironmentName specifies the [Azure DevOps environment] ##(https://learn.microsoft.com/en-us/azure/devops/pipelines/process/environments)## that's being deployed to enable tracking deployment history and set permissions and approvals for deployment to specific environments. Set this value to Validate, Test, or Production, as appropriate."

appreciate your patience, where am I going wrong?
Is it best to have a call or something?
Thanks again

**Comment by mikefactorial** [comment]
@ChristopheHare if the deployment pipelines aren't being created it's possible that there is an error being swallowed somewhere in the Update Deployment Settings step of the export pipeline. You should be able to search the log for that step for the word "error" or something similar to see if this is the case. More than likely it's a permissions issue, but if you find an error occurring feel free to forward it and we can assist with troubleshooting.

**Comment by ChristopheHare** [comment]
Thanks Mike, will take a look :)

**Comment by ChristopheHare** [comment]
@mikefactorial  i have found error 404 Not Found. within the log, what's the best way to forward the log to yourself?
